fw/cui_amt22.h

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 mjbots Robotic Systems, LLC.  info@mjbots.com
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "mbed.h"
+
+#include "hal/spi_api.h"
+
+#include "fw/ccm.h"
+#include "fw/moteus_hw.h"
+#include "fw/stm32_spi.h"
+
+const uint32_t AMT22_TIME_BETWEEN_READS = 40;
+const uint32_t AMT22_TIME_BETWEEN_BYTES = 3;
+const uint32_t AMT22_TIME_AFTER_CS = 3;
+
+namespace moteus {
+
+class CuiAmt22 {
+ public:
+  using Options = Stm32Spi::Options;
+
+  CuiAmt22(const Options& options, MillisecondTimer* timer)
+      : spi_([&]() {
+               auto options_copy = options;
+               options_copy.width = 8;
+               options_copy.mode = 0;
+               return options_copy;
+             }()),
+        cs_(std::in_place_t(), options.cs, 1),
+        timer_(timer) {
+  }
+
+  uint16_t Sample() MOTEUS_CCM_ATTRIBUTE {
+    ISR_StartSample();
+    uint32_t value = 0;
+    uint32_t timeout = 2000;
+    while (!ISR_Update(&value) && timeout--) {}
+    return value;
+  }
+
+  void ISR_StartSample() MOTEUS_CCM_ATTRIBUTE {
+    if (state_ == 4) {
+      state_ = 0;
+    }
+  }
+
+  // return value: boolean whether or not we finished reading the sensor and updated the value
+  int ISR_Update(uint32_t* value_ptr) MOTEUS_CCM_ATTRIBUTE {
+    const uint32_t now_us = timer_->read_us();                
+    const uint32_t delta_us = (now_us - last_byte_finished_us_);
+    uint16_t value = 0;
+    switch (state_) {
+      case 0:
```

Comment:
`case` statements should have braces around them: https://google.github.io/styleguide/cppguide.html#Switch_Statements